build_tools/style.fish

```diff
@@ -1,38 +1,54 @@
 #!/usr/bin/env fish
 #
-# This runs C++ files and fish scripts (*.fish) through their respective code
-# formatting programs.
+# This runs Python files, fish scripts (*.fish), and Rust files
+# through their respective code formatting programs.
 #
+# `--all`: Format all eligible files instead of the ones specified as arguments.
+# `--check`: Instead of reformatting, fail if a file is not formatted correctly.
+# `--force`: Proceed without asking if uncommitted changes are detected.
+#            Only relevant if `--all` is specified but `--check` is not specified.
+
 set -l fish_files
 set -l python_files
 set -l rust_files
 set -l all no
 
-if test "$argv[1]" = --all
+argparse all check force -- $argv
+or exit $status
+
+if set -l -q _flag_all
     set all yes
-    set -e argv[1]
+    if set -q argv[1]
+        echo "Unexpected arguments: '$argv'"
+        exit 1
+    end
 end
 
-if set -q argv[1]
-    echo "Unexpected arguments: '$argv'"
-    exit 1
-end
+set -l repo_root (status dirname)/..
 
 if test $all = yes
-    set -l files (git status --porcelain --short --untracked-files=all | sed -e 's/^ *[^ ]* *//')
-    if set -q files[1]
-        echo
-        echo 'You have uncommitted changes. Are you sure you want to restyle?'
-        read -P 'y/N? ' -n1 -l ans
-        if not string match -qi y -- $ans
-            exit 1
+    if not set -l -q _flag_force; and not set -l -q _flag_check
+        # Potential for false positives: Not all fish files are formatted, see the `fish_files`
+        # definition below.
+        set -l relevant_uncommitted_changes (git status --porcelain --short --untracked-files=all | sed -e 's/^ *[^ ]* *//' | grep -E '.*\.(fish|py|rs)$')
+        if set -q relevant_uncommitted_changes[1]
+            for changed_file in $relevant_uncommitted_changes
+                echo $changed_file
+            end
+            echo
+            echo 'You have uncommitted changes (listed above). Are you sure you want to restyle?'
+            read -P 'y/N? ' -n1 -l ans
+            if not string match -qi y -- $ans
+                exit 1
+            end
         end
     end
-    set fish_files share/**.fish
-    set python_files {doc_src,share,tests}/**.py
-    set rust_files fish-rust/src/**.rs
+    set fish_files $repo_root/{benchmarks,build_tools,etc,share}/**.fish
+    set python_files $repo_root/**.py
```

Comment:
not sure but I might keep python filter (`{doc_src,share,tests}`) because globbing the entire tree feels somehow wrong.. ideally our build system doesn't need to glob but has an explicit list of all source files..
I guess we're unlikely to add python files elsewhere

Actually I have these, which I'd prefer not to format:

```
would reformat build_tools/../t/test.py
would reformat build_tools/../t/test3.py
would reformat build_tools/../t/tmp.py
would reformat build_tools/../t/gdb.py
```